.github/workflows/test.yml

```diff
@@ -0,0 +1,32 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+# All jobs have the same outcome. We define multiple for resiliency reasons.
+jobs:
+  # In thechangelog/pipedream repository (a.k.a. upstream),
+  # this is the preferred default:
+  on-namespace:
+    if: ${{ contains(vars.RUNS_ON, 'namespace') }}
+    uses: ./.github/workflows/_namespace.yml
+    secrets: inherit
+
+  # Just in case Namespace.so misbehaves, we want a fallback:
+  # Always run two of everything™
+  on-github-fallback:
+    needs: on-namespace
+    if: ${{ failure() }}
+    uses: ./.github/workflows/_github.yml
+    secrets: inherit
```

Comment:
Under what conditions might `ubuntu-22.04` succeed when `nscloud-ubuntu-24.04-amd64-2x4` has failed (not skipped)?

Reply:
What a great catch! It was meant to be Ubuntu `24.04` in all places.

The OS version was meant to be the same. The only delta should be the runtime: GitHub vs Namespace.so.